2.4/root/usr/bin/run-mongod

```diff
@@ -46,8 +32,8 @@ function cleanup() {
     mongo_remove
   fi
   echo "=> Shutting down MongoDB server ..."
-  if [ -f "${MONGODB_PID_FILE}" ]; then
-    kill -2 $(cat ${MONGODB_PID_FILE})
+  if [[ -s ${MONGODB_DATADIR}/mongod.lock ]]; then
+    kill -2 $(cat ${MONGODB_DATADIR}/mongod.lock)
```

Comment:
What "backward-compat" do you mean?

`MONGODB_PID_FILE` before this PR is specified in `/usr/share/container-scripts/mongodb/common.sh` and it is used to stop container. In this PR `MONGODB_PID_FILE` is not used in scripts, so no need have it present in scripts.
However if you know some usecase when MONGODB_PID_FILE should be created with default config file. It could be easily added back to config.tepmlate.